### PR TITLE
Feature/ecs 71 ecs naming convention alignment

### PR DESCRIPTION
**Pull request by mhennahane**
Comments on when to use "ecs", "lambda", "sns" as the AWS service in the name are welcome.

As is feedback on name choices.

### CODE REVIEW

**Comment by mhennahane** [comment]
NB: I added one more commit after original PR - lambda function names were not conforming.

File: graceful_shutdown.tf [hunk 1]
```diff
@@ -174,8 +176,8 @@ resource "aws_lambda_function" "asg_drain" {
 }
 
 resource "aws_sns_topic" "lambda_asg_drain" {
-  name         = "lambda_asg_drain-${var.name}"
-  display_name = "ASG Drain Lifecycle Hook"
+  name         = "${var.name}-sns-asg-drain"
```
Comment (watchwithmike) [inline graceful_shutdown.tf]: `-sns-` not needed and I'd probably make it `-asg-draining`

File: graceful_shutdown.tf [hunk 2]
```diff
@@ -213,7 +215,7 @@ resource "aws_sns_topic_subscription" "lambda_asg_drain_subscription" {
 }
 
 resource "aws_lambda_permission" "allow_sns_to_call_asg_drain_lambda" {
-  statement_id  = "AllowExecutionFromSNS-${var.name}"
+  statement_id  = "${var.name}-lambda-allow-sns-asg-drain"
```
Comment (watchwithmike) [inline graceful_shutdown.tf]: Is this a policy?

File: main.tf [hunk 3]
```diff
@@ -109,7 +109,7 @@ resource "aws_autoscaling_group" "ecs" {
 
   tags = [{
     key                 = "Name"
```
Comment (watchwithmike) [inline main.tf]: Can we add "Environment" as a tag here?

Comment (watchwithmike) [reply main.tf]: Maybe that's what I see down below...

File: autoscaling.tf [hunk 4]
```diff
@@ -1,61 +1,61 @@
...
   comparison_operator = "LessThanThreshold"
-  evaluation_periods = "1"
-  metric_name = "SchedulableContainers"
-  namespace = "WW/ECS"
-  period = "60"
-  statistic = "Average"
-  threshold = "1"
+  evaluation_periods  = "1"
+  metric_name         = "SchedulableContainers"
+  namespace           = "WW/ECS"
```
Comment (watchwithmike) [inline autoscaling.tf]: Not sure how this is used.

File: main.tf (hunk 3 above)
Comment (mhennahane) [reply main.tf]: Environment tag IS what you saw on the next line.

File: autoscaling.tf (hunk 4 above)
Comment (mhennahane) [reply autoscaling.tf]: This is a custom metric that we use to determine when to scale.  SchedulableContainers is how much headroom do we have - i.e. how many containers of the largest size could we launch right now.   If it's large - i.e. much headroom - we scale back.   If it's low - i.e. no headroom - we spin up more EC2 instances.

File: graceful_shutdown.tf (hunk 2 above)
Comment (mhennahane) [reply graceful_shutdown.tf]: It's a lambda permission.   Like a policy but not technically one.

File: graceful_shutdown.tf (hunk 1 above)
Comment (mhennahane) [reply graceful_shutdown.tf]: This is an SNS topic, so I named it -sns-asg-drain.

Comment (mhennahane) [reply graceful_shutdown.tf]: Just make a final decision - leave or change - and we can put this to bed.

Comment (watchwithmike) [reply graceful_shutdown.tf]: Our SNS topic naming convention doesn't include -sns- so let's just go with:
`name         = "${var.name}-asg-draining"`

File: graceful_shutdown.tf (hunk 2 above)
Comment (watchwithmike) [reply graceful_shutdown.tf]: Let's leave this as-is

Comment (mhennahane) [reply graceful_shutdown.tf]: OK, left

File: graceful_shutdown.tf (hunk 1 above)
Comment (mhennahane) [reply graceful_shutdown.tf]: This is changed in latest commit.

**Comment by mhennahane** [comment]
All changes should be addressed.

File: autoscaling.tf (hunk 4 above)
Comment (watchwithmike) [reply autoscaling.tf]: So, are we leaving it as `WW/ECS`?  I don't like that, can we just use `Watchwith`?

Comment (mhennahane) [reply autoscaling.tf]: My bad - I made all the changes in the UserData, but this one slipped through.  I have a call right now, but the rest of the day is yours.

Comment (mhennahane) [reply autoscaling.tf]: BTW, we chose "Watchwith <env>"

**Pull request merged**